src/coord/src/coord.rs

```diff
@@ -1200,6 +1200,25 @@ impl Coordinator {
             .filter(|(_, frontier)| frontier != &Antichain::new())
             .collect();
 
+        for (id, frontier) in since_updates.iter() {
+            let item = self.catalog.try_get_by_id(*id).map(|e| e.item());
+            if let Some(CatalogItem::Index(catalog::Index { on, .. })) = item {
+                if let CatalogItem::Table(catalog::Table {
+                    persist: Some(persist),
+                    ..
+                }) = self.catalog.get_by_id(on).item()
+                {
+                    if self.catalog.default_index_for(*on) == Some(*id) {
+                        // TODO: more idiomatic antichain usage.
+                        let _ = persist.write_handle.allow_compaction(frontier[0]);
+                        // TODO: its not clear to me whether we need to await the
```

Comment:
Not doing something with the response scares me. It's already proves useful (at the very least to philip) to be printing errors when these things fail for seal (which also doesn't await anything in the critical path), so I think we should do the same here

Reply:
What could we do that's useful? Printing a log message seems not beneficial. This isn't in a user path so there's nothing to tell them. It doesn't affect correctness to ignore compaction.

Reply:
I don't have a strong feeling. I just noted that this was the precedent set by the coordinator in how it gives since updates to the dataflow workers. Happy to defer to whatever you both decide on.

Reply:
I'm not sure that's a relevant precedent. The insert, seal, and allow_compaction messages that coord sends to dataflow are all infallible, but the persist equivalents are not and that's a pretty big difference.

I've been trained for years and years to never ignore errors and it's served me well. Logging this one isn't that actionable by a user (maybe? if they tell us about these errors that's actually a good outcome), but it's better than silently ignoring it. We could have a counter for failures, which might be the right answer in the long run, but in the short term I think that's even less likely to get looked at than logs.

> It doesn't affect correctness to ignore compaction.

Possibly nitpicky, but I'm not sure I agree with this. Ignoring an individual compaction message doesn't affect correctness, agreed, but we "guarantee" (intend to guarantee) an amount of disk usage somehow proportional to the size of the non-retracted records in a collection. If we ignore sustained failure in compaction messages, then we won't be living up to that.

On the other hand, given the indirectness of the id-hopping we're doing here, it doesn't really inspire confidence that we're actually compacting all the tables even if nothing is erroring. Seems like an easy codepath to accidentally break. So perhaps there's some larger thing we need to be doing here to be ensuring that everything is healthy (monitor the compaction frontier of all active persisted collections, for example).

My strong inclination is to make this a log message with a TODO to figure out something smarter. If matt is uncomfortable with that, then just the TODO is fine for now.

Reply:
sgtm